agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannelTest.java

```diff
@@ -65,18 +62,20 @@
 
 public class TelemetryChannelTest {
   RecordingHttpClient recordingHttpClient;
-  private static final AtomicInteger requestCount = new AtomicInteger();
+  private final Cache<String, String> ikeyRedirectCache =
+      Cache.newBuilder().setMaximumSize(5).build();
   private static final String INSTRUMENTATION_KEY = "00000000-0000-0000-0000-0FEEDDADBEEF";
   private static final String REDIRECT_INSTRUMENTATION_KEY = "00000000-0000-0000-0000-0FEEDDADBEEE";
   private static final String END_POINT_URL = "http://foo.bar";
   private static final String REDIRECT_URL = "http://foo.bar.redirect";
 
   @TempDir File tempFolder;
 
-  private TelemetryChannel getTelemetryChannel(boolean followInstrumentationKeyForRedirect)
+  private TelemetryChannel getTelemetryChannel(@Nullable Cache<String, String> ikeyRedirectCache)
       throws MalformedURLException {
     List<HttpPipelinePolicy> policies = new ArrayList<>();
-    policies.add(new RedirectPolicy(followInstrumentationKeyForRedirect));
+
+    policies.add(new RedirectPolicy(ikeyRedirectCache));
```

Comment:
then can remove `ikeyRedirectCache` everywhere I think
```suggestion
    policies.add(new RedirectPolicy(Cache.newBuilder().setMaximumSize(5).build()));
```